data-validation/check-input-datasets.js

```diff
@@ -0,0 +1,129 @@
+const fsPromises = require("fs").promises;
+const Ajv = require("ajv").default;
+
+const { readDatasetJson, readAndParseFile, readPossibleZippedFile } = require("../utils");
+const dataPrep = require("./data-prep");
+
+// ref schemas
+const refSchemas = [
+    require("./schema/definitions.schema.json"),
+    require("./schema/discrete-feature-options.schema.json"),
+    require("./schema/array-items/discrete-feature-option.schema.json"),
+    require("./schema/array-items/file-info.schema.json"),
+    require("./schema/array-items/input-measured-features.schema.json"),
+];
+
+const inputMegaset = require("./schema/input-megaset.schema.json");
+const inputDatasetInfo = require("./schema/input-dataset-info.schema.json");
+const inputMeasuredFeatures = require("./schema/input-measured-features-doc.schema.json");
+const inputDataSet = require("./schema/input-dataset.schema.json");
+const inputImages = require("./schema/images.schema.json");
+const featureDef = require("./schema/feature-def.schema.json");
+
+const ajv = new Ajv({
+    coerceTypes: true,
+    removeAdditional: true,
+    schemas: [
+        ...refSchemas,
+        inputImages,
+        featureDef,
+        inputMegaset,
+        inputDatasetInfo,
+        inputMeasuredFeatures,
+        inputDataSet,
+    ],
+});
+
+const checkForError = (fileName, json, schemaFileName) => {
+    const { valid, error } = dataPrep.validate(
+        json,
+        ajv.getSchema(schemaFileName)
+    );
+    if (!valid) {
+        console.log("\x1b[0m", `${fileName}`, "\x1b[31m", `failed because: ${JSON.stringify(error)}`);
+        return true;
+    } else {
+        console.log("\x1b[0m", `${fileName}: check against ${schemaFileName}`, "\x1b[32m", "passed");
+        return false;
+    }
+};
+
+const unpackInputDataset = async (datasetReadFolder) => {
+    const datasetJson = await readDatasetJson(datasetReadFolder);
+    const featureDefs = await readAndParseFile(
+        `${datasetReadFolder}/${datasetJson.featureDefsPath}`
+    );
+    const images = await readAndParseFile(
+        `${datasetReadFolder}/${datasetJson.viewerSettingsPath}`
+    );
+    const measuredFeatures = await readPossibleZippedFile(datasetReadFolder, datasetJson.featuresDataPath)
+
+    const inputDataset = {
+        dataset: datasetJson,
+        "feature-defs": featureDefs,
+        "measured-features": measuredFeatures,
+        images: images,
+    };
+    return inputDataset;
+}
+
+const validateDatasets = () => {
+    fsPromises.readdir("./").then(async (files) => {
```

Comment:
would be nice if all the actual data were in a specific subdir like `data\`.  I made a new gh ticket for that